pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -690,6 +691,75 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 
 	switch {
 	case cd.Spec.Provisioning != nil:
+		// Ensure the install config matches the ClusterDeployment:
+		// TODO: Move with the openshift-installer ClusterInstall code when we implement https://issues.redhat.com/browse/HIVE-1522
+		if cd.Spec.Provisioning.InstallConfigSecretRef == nil {
+			cdLog.Info("install config not specified")
+			conditions, changed := controllerutils.SetClusterDeploymentConditionWithChangeCheck(
+				cd.Status.Conditions,
+				hivev1.RequirementsMetCondition,
+				corev1.ConditionFalse,
+				"InstallConfigRefNotSet",
+				"Install config reference is not set",
+				controllerutils.UpdateConditionIfReasonOrMessageChange)
```

Comment:
Is the InstallConfigSecretRef not optional? From what I can tell from code - it is passed through to the installer if present

Reply:
It's optional at the API level, but we can't go further without one here. If we're doing a normal openshift-install provision, you must supply one, and if we don't have it there's no point in launching the ClusterProvision. Previously we'd just let kube hold the pod up because it's missing a secret to mount.